ietf-optical-impairment-topology.tree

```diff
@@ -49,39 +56,110 @@ module: ietf-optical-impairment-topology
+    |        +--ro channel-input-power?      dbm-t
+    |        +--ro total-input-power?        dbm-t
+    +--ro transponder-list* [transponder-id]
+       +--ro transponder-id      uint32
+       +--ro transceiver-list* [transceiver-id]
+          +--ro transceiver-id     uint32
+          +--ro supported-modes
+             +--ro supported-mode* [mode-id]
+                +--ro mode-id                      string
+                +--ro (mode)
+                   +--:(G.698.2)
+                   |  +--ro standard-mode?         standard-mode
+                   +--:(organizational-mode)
+                   |  +--ro organizational-mode
+                   |     +--ro operational-mode?
+                   |     |       operational-mode
+                   |     +--ro organization-identifier?
+                   |     |       organization-identifier
+                   |     +--ro min-central-frequency?
+                   |     |       frequency-thz
+                   |     +--ro max-central-frequency?
+                   |     |       frequency-thz
+                   |     +--ro minimum-channel-spacing?
+                   |     |       frequency-ghz
+                   |     +--ro min-channel-output-power?   dbm-t
+                   |     +--ro max-channel-output-power?   dbm-t
+                   |     +--ro min-channel-input-power?    dbm-t
+                   |     +--ro max-channel-input-power?    dbm-t
+                   |     +--ro max-total-input-power?      dbm-t
+                   +--:(explicit-mode)
+                      +--ro explicit-mode
+                         +--ro supported-modes
+                         |  +--ro supported-application-codes*
+                         |  |       -> ../../mode-id
+                         |  +--ro supported-organizational-modes*
+                         |          -> ../../mode-id
+                         +--ro line-coding-bitrate?
+                         |       identityref
+                         +--ro max-polarization-mode-dispersion?
+                         |       decimal64
+                         +--ro max-chromatic-dispersion?
+                         |       decimal64
+                         +--ro chromatic-and-polarization-dispersion-penalty* []
+                         |  +--ro chromatic-dispersion
+                         |  |       decimal64
+                         |  +--ro polarization-mode-dispersion
+                         |  |       decimal64
+                         |  +--ro penalty
+                         |          decimal64
+                         +--ro max-diff-group-delay?
+                         |       int32
+                         +--ro max-polarization-dependent-loss?
+                         |       decimal64
+                         +--ro available-modulation-type?
+                         |       identityref
+                         +--ro min-OSNR?
+                         |       int32
```

Comment:
should probably have 1-2 decimal digits .e. fraction-digits 2 (tbd)

Reply:
In optical-impairments we have already e.g. roadm-osnr , defined with "fraction-digit 5" . Should be all aligned osnr at the same fraction-digit ? In case of transponder the definiton is on layer0-type-ext.

Reply:
Let's define a type for OSNR and use it in any cases you need.

Reply:
At the moment in the model we have different definition of measure unit for SNR:
       type decimal64 {
          fraction-digits 5;
        }
        units "dB"; 
or
      type decimal64 {
        fraction-digits 5;
      }
      units "dB@0.1nm";
or
    leaf  min-OSNR {
      type int32;
      units "dB";
      config false;
      description "min OSNR measured over 12.5 GHz
      resolution bandwidth:

What is the common reference to be defined ?

Reply:
I agree to introduce a "SNR" type which embeds the proper units.

Units should be "dB" but we must specify the resolution bandwidth in the description. I see two options:

1. description "(Optical) Signal to Noise Ratio measured over 12.5 GHz resolution bandwidth"
2. description "(Optical) Signal to Noise Ratio measured over 0.1 nm resolution bandwidth"

Though the former is a little more precise  the latter is far more widely used, thus probably we should go for option 2. For example see definition 5.1.19 in ITU-T Rec. G.661 (07/2007) .

Alternatively, we can specify the resolution bandwidth in the units, as "dB [0.1 nm]" or "dB @ 0.1nm"

Reply:
I think Option 2 can be used also with unit resolution. My proposal at the end is:
typedef snr {
    type decimal64 {
      fraction-digits 2;
    }
    units "dB@0.1nm";
    description
    "(Optical) Signal to Noise Ratio measured over 0.1 nm resolution bandwidth";
  }

Reply:
I also prefer to have the resolution right in the units field. 
`dB@0.1nm `
is OK for me
type should not be integer but decimal64 as you state. 2 digit looks enough from my point of view...